ogre2/src/Ogre2LidarVisual.cc

```diff
@@ -297,6 +297,11 @@ void Ogre2LidarVisual::Update()
       }
     }
 
+    // The newly created dynamic lines are having default visibility as true.
+    // The visibility needs to be set as per the current value after the new
+    // renderables are created.
+    this->SetVisible(this->Visible());
```

Comment:
I think you can move this call out side of the for loop, e.g. at the end of the function? Same for `OgreLidarVisual` class